library/src/blas3/rocblas_herk_batched.cpp

```diff
@@ -0,0 +1,200 @@
+/* ************************************************************************
+ * Copyright 2016-2020 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#include "logging.h"
+#include "rocblas_herk.hpp"
+#include "utility.h"
+
+namespace
+{
+    template <typename>
+    constexpr char rocblas_herk_name[] = "unknown";
+    template <>
+    constexpr char rocblas_herk_name<rocblas_float_complex>[] = "rocblas_cherk_batched";
+    template <>
+    constexpr char rocblas_herk_name<rocblas_double_complex>[] = "rocblas_zherk_batched";
+
+    template <typename T, typename U>
```

Comment:
Why two template parameters?  You should use one argument and `real_t<T>` for the real part.

We cannot support arbitrary mixed precision without a completely different implementation and tests, and so adding flexibility for the sake of flexibility is just confusing. Stick to one main type, and derive related types using traits like `real_t<T>`